recurring_contract/model/recurring_contract.py

```diff
@@ -311,11 +360,81 @@ def _compute_next_invoice_date(self, contract):
 
         return next_date
 
+    def _update_invoice_lines(self, cr, uid, contract, invoice_ids,
+                              context=None):
+        """Update invoice lines generated by a contract, when the contract
+        was modified and corresponding invoices were cancelled.
+
+        Parameters:
+            - invoice_ids (list): ids of draft invoices to be
+                                  updated and validated
+        """
+        invoice_obj = self.pool.get('account.invoice')
+        inv_line_obj = self.pool.get('account.invoice.line')
+        group_obj = self.pool.get('recurring.contract.group')
+        for invoice in invoice_obj.browse(cr, uid, invoice_ids, context):
+            # Update payment term and generate new invoice_lines
+            invoice.write({
+                'payment_term': contract.group_id.payment_term_id and
+                contract.group_id.payment_term_id.id or False})
+            old_lines_ids = [invl.id for invl in invoice.invoice_line
+                             if invl.contract_id.id == contract.id]
+            inv_line_obj.unlink(cr, uid, old_lines_ids)
+            context['no_next_date_update'] = True
+            group_obj._generate_invoice_lines(cr, uid, contract,
+                                              invoice.id, context)
+            del(context['no_next_date_update'])
+
+    def _on_change_next_invoice_date(
+            self, cr, uid, ids, new_invoice_date, context=None):
+        for contract in self.browse(cr, uid, ids, context):
+            if contract.next_invoice_date:
+                next_invoice_date = datetime.strptime(
+                    contract.next_invoice_date, DF)
+                new_invoice_date = datetime.strptime(new_invoice_date, DF)
+                if (next_invoice_date > new_invoice_date):
+                    raise orm.except_orm(
+                        'Error',
+                        _('You cannot set the next invoice date'

```

Comment:
Be more specific in error message by telling we cannot rewind next_invoice_date.